.github/workflows/build.yml

```diff
@@ -169,3 +169,69 @@ jobs:
         env:
           NO_MINETEST_GAME: 1
           NO_PACKAGE: 1
+  msvc:
+    name: VS 2019 ${{ matrix.config.arch }}-${{ matrix.type }}
+    runs-on: windows-2019 
+    env:
+      VCPKG_VERSION: c7ab9d3110813979a873b2dbac630a9ab79850dc
+#                    2020.04
+      vcpkg_packages: irrlicht zlib curl[winssl] openal-soft libvorbis libogg sqlite3 freetype luajit
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - {
+              arch: x86,
+              generator: "-G'Visual Studio 16 2019' -A Win32",
+              vcpkg_triplet: x86-windows
+            }
+          - {
+              arch: x64,
+              generator: "-G'Visual Studio 16 2019' -A x64",
+              vcpkg_triplet: x64-windows
+            }
+        type: [portable, installer]
```

Comment:
`portable` is pretty much guaranteed to work, it's enough if these jobs test only the installer

Reply:
Someone still have to do it. If there is a change, that cant be tested by simply compiling it, users have to either compile the changes by themself or they can now just download the package and and try it without the need of compiling. 

It runs parallel, its already implemented, why not use it?

Reply:
> Someone still have to do it.

The Gitlab pipeline already tests ZIP package generation.

> If there is a change, that cant be tested by simply compiling it, users have to either compile the changes by themself or they can now just download the package and and try it without the need of compiling.

I am not saying artifacts should be removed.

> It runs parallel, its already implemented, why not use it?

I don't really like have the CI waste time even if it's "free".

Reply:
remember that we can run 10 jobs in parallel in the free version here, but we may just run this on master, it's less runs that on each pull request, and is more reasonable no ? :)

Reply:
There's only 10 jobs in parallel? Then this definitely does make a difference.

Reply:
There are 20 concurrent jobs: https://help.github.com/en/actions/getting-started-with-github-actions/about-github-actions#usage-limits

Gitlab only compiles the master branch, not the PR.

As long as there is a working artifact it is not "waste time" imo.
----
If you really want to save some jobs on the PR, I would suggest to skip the installer builds on PR. Potential testers can then have the stable release version installed, but can still test the portable version without breaking one of their worlds. Would that be ok?

Reply:
oh yeah it's 20, we are fine then :) sorry i misread the doc at the beginning :)

Reply:
> Potential testers can then have the stable release version installed, but can still test the portable version without breaking one of their worlds. Would that be ok?

Yeah skip the `installer` variant then.

Reply:
> oh yeah it's 20, we are fine then

> Yeah skip the `installer` variant then.

😂 Keep it as its currently is, or skip the installer?

Reply:
I still propose having just `portable` builds since that fulfills both the "test compiling with MSVC" and "produce usable artifact".

Reply:
Can you produce installer builds on tagged commits?

Reply:
> Can you produce installer builds on tagged commits?
could be possible https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idif But it depends on how does the if work in combination with the matrix

Reply:
> 
> 
> > Can you produce installer builds on tagged commits?
> 
> could be possible https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idif But it depends on how does the if work in combination with the matrix

Sadly it isn't possible yet https://github.community/t5/GitHub-Actions/Matrix-cannot-be-used-in-jobs-level-if/td-p/41048

I would suggest to create a build-release.yml that is doing the full job (including minetest game) if a tag is pushed. But I think that will be better done in a separate PR

I currently disabled the installer builds. If someone is working on the installer he can simply enable them.